core/storage/base_model/gae_models.py

```diff
@@ -1496,6 +1501,46 @@ def get_version(
                 return None
             raise e
 
+    @classmethod
+    def get_version_multi(
+        cls: Type[SELF_VERSIONED_MODEL],
+        entity_ids_versions: List[Tuple[str, Optional[int]]]
+    ) -> List[Optional[SELF_VERSIONED_MODEL]]:
+        entity_ids = set([entity_id for (entity_id, _) in entity_ids_versions])
+        current_version_models = cls.get_multi(
+            entity_ids, include_deleted=False)
+        for current_version_model in current_version_models:
+            current_version_model._require_not_marked_deleted()  # pylint: disable=protected-access
+        latest_versions_by_id = {
+            current_version_model.id: current_version_model.version
+            for current_version_model in current_version_models}
+
+        # The assumption is that every instance of cls uses the same
+        # SNAPSHOT_CONTENT_CLASS, so only one fake instance is needed to fetch
+        # that class for a get_multi() operation.
+        temp_model = cls(id='', version_number=0)
```

Comment:
How come we can't just do cls.SNAPSHOT_CONTENT_CLASS? Not sure why we need a temp model here.

Reply:
Uh, my Python knowledge has greatly degraded over the years. :) I assumed that this was an instance property, not a class property, as other uses are also referencing it from a member rather than the class. Is that not correct?

Reply:
It's defined as a class property in VersionedModel so I believe you can access it from the class as well, similar to COMMIT_TYPE_CHOICES.